### PR TITLE
Fix asa app test endpoint access failed and support the web app refresh

### PR DESCRIPTION
**Pull request by moarychan**
## Purpose
- Fix asa app test endpoint access failed
- Support the web app refresh

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

## What to Check
Test manually.

## Other Information
N/A

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import React, { useReducer, FC } from 'react';
-import { BrowserRouter } from 'react-router-dom';
+import React, {useReducer, FC, useState} from 'react';
+import { HashRouter } from 'react-router-dom';
```
Comment (saragluna) [inline client/src/App.tsx:2]: what's the difference between these two?

File: client/src/App.tsx [hunk 2]
```diff
@@ -11,20 +11,29 @@ import { ThemeProvider } from '@fluentui/react';
 import Telemetry from './components/telemetry';
 
 export const App: FC = () => {
+  const [listId, setListId] = useState("")
```
Comment (saragluna) [inline client/src/App.tsx:14]: What's this useState for?

File: client/src/config/index.ts [hunk 3]
```diff
@@ -13,11 +13,11 @@ export interface AppConfig {
 
 const config: AppConfig = {
     api: {
-        baseUrl: '' //window.ENV_CONFIG.REACT_APP_API_BASE_URL || 'http://localhost:3100'
+        baseUrl: window.ENV_CONFIG.REACT_APP_API_BASE_URL || ''
```
Comment (saragluna) [inline client/src/config/index.ts:16]: Why did you remove this localhost?

File: client/src/App.tsx (hunk 1 above)
Comment (moarychan) [reply client/src/App.tsx:2]: In the past, using the BrowserRouter method would change the browser's address, and the changed address happened to be the SB application backend REST API address, so after refreshing, the browser's response became the json response of the REST API; the HashRouter will not change the browser's address.

File: client/src/App.tsx (hunk 2 above)
Comment (moarychan) [reply client/src/App.tsx:14]: It's a global getter and setter for current selected list id, when this value changed, it will effect other dependencies, keep updated data in sync.

File: client/src/config/index.ts (hunk 3 above)
Comment (moarychan) [reply client/src/config/index.ts:16]: I assume the 'http://localhost:3100' is only used when the client module project is ran in local, and empty '' can also work, uncomment the environment to adapter other customer scenarios.
